api-test/internal/e2e/e2e.go

```
@@ -0,0 +1,731 @@
+		}
+	}
+	return r.client
+}
+
+// do performs an HTTP call and records it (Authorization redacted) into calls.
+func (r *Runner) do(calls *[]result.HTTPCall, label, method, u string, headers map[string]string, body string) (int, []byte, error) {
+	var rdr io.Reader
+	if body != "" {
+		rdr = bytes.NewBufferString(body)
+	}
+	req, err := http.NewRequest(method, u, rdr)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)